paas/_base.j2

```diff
@@ -2,10 +2,12 @@
 
 applications:
   - name: {{ app|replace('_', '-') }}-release
+{% if routes|default([]) %}
```

Comment:
Do we need the default here? I don't think it adds anything when we're doing an `if` check; only needed when iterating with `for`, I'd have thought.

Reply:
Because the Jinja environment is set up in `StrictUndefined` mode in the generate paas manifest script (http://jinja.pocoo.org/docs/2.9/api/#jinja2.StrictUndefined), the only operation allowed on an undefined object is to check if it is defined. Therefore, if `routes` is not defined and you try `if routes` then we will throw an exception. Therefore we need to use the `default` filter (https://github.com/pallets/jinja/blob/master/jinja2/filters.py#L355) I believe.

Reply:
Maybe what we want to actually check then is that it's defined? i.e `if routes is defined`?